src/process.cr

```diff
@@ -399,6 +407,14 @@ class Process
   end
 
   private def channel
+    if channel = @channel
+      channel
+    else
+      raise "BUG: Notification channel was not initialized for this process"
+    end
+  end
```

Comment:
This method is only used for `Process#wait`.
And even in its method, it could be `ex = channel.try &.receive`.
I propose to remove this method or move the logic to `Process#wait`.